crates/sui-benchmark/src/workloads/adversarial.rs

```diff
@@ -69,16 +89,31 @@ impl Payload for AdversarialTestPayload {
     }
 
     fn make_transaction(&mut self) -> VerifiedTransaction {
-        // TODO: default benchmarking gas coins are too small to use MAX_TX_GAS. But we will want to be able to use that much to hit some limits
-        let gas_budget = MAX_TX_GAS / 100;
-        // TODO: generate random number, convert it to a PayloadType, call the appropriate function
+        let payload_type: AdversarialPayloadType = rand::random();
+
+        let gas_budget = self.system_state_observer.protocol_config.max_tx_gas();
+        let mut payload_args =
+            get_payload_args(&payload_type, &self.system_state_observer.protocol_config);
+
+        if let AdversarialPayloadType::DynamicFieldsCount = payload_type {
```

Comment:
This is a bit funky--I would suggest having `get_payload_args` take `self` and adding the shared object arg inside there instead. This will also work better for future workloads with other special args.